File: scratch/analyze_quyche.js
const fs = require('fs');
const path = require('path');

// Read file in UTF-16LE
const filePath = 'd:\\QuocAnh\\2026\\01.Project\\qlda-ddcn-ht\\Quyche.md';
const content = fs.readFileSync(filePath, 'utf16le');

// Clean up carriage returns
const lines = content.replace(/\r\n/g, '\n').split('\n');
console.log(`Total lines in Quyche.md: ${lines.length}`);

const chapters = [];
const articles = [];

lines.forEach((line, idx) => {
    const lineClean = line.trim();
    if (!lineClean) return;

    // Check for chapter
    if (/\*\*Chương\s+[IVXLCDM]+\*\*/i.test(lineClean) || /^Chương\s+[IVXLCDM]+/i.test(lineClean)) {
        chapters.push({ lineNum: idx + 1, text: lineClean });
    }
    // Check for article
    else if (/\*\*Điều\s+\d+/i.test(lineClean) || /^Điều\s+\d+/i.test(lineClean)) {
        articles.push({ lineNum: idx + 1, text: lineClean });
    }
});

console.log("\n--- CHAPTERS FOUND ---");
chapters.forEach((chap) => {
    console.log(`Line ${chap.lineNum}: ${chap.text}`);
    // Print the next non-empty lines
    const nextLines = [];
    let i = chap.lineNum;
    while (i < lines.length && nextLines.length < 2) {
        if (lines[i].trim()) {
            nextLines.push(lines[i].trim());
        }
        i++;
    }
    console.log(`  Title lines:`, nextLines);
});

console.log(`\n--- ARTICLES FOUND (Total: ${articles.length}) ---`);
articles.forEach((art, idx) => {
    if (idx < 10 || idx >= articles.length - 5) {
        console.log(`Line ${art.lineNum}: ${art.text}`);
    } else if (idx === 10) {
        console.log("...");
    }
});

File: scratch/analyze_quyche.py
# -*- coding: utf-8 -*-
import re

with open(r"d:\QuocAnh\2026\01.Project\qlda-ddcn-ht\Quyche.md", "r", encoding="utf-16-le") as f:
    content = f.read()

# Let's clean up any weird BOM or carriage returns
content = content.replace('\r\n', '\n')

lines = content.split('\n')
print(f"Total lines: {len(lines)}")

chapters = []
articles = []

for idx, line in enumerate(lines):
    line_clean = line.strip()
    if not line_clean:
        continue
    # Look for Chapter headings (e.g. **Chương I**, **Chương II**, etc.)
    if re.search(r'\*\*Chương\s+[IVXLCDM]+\*\*', line_clean, re.IGNORECASE) or re.search(r'^Chương\s+[IVXLCDM]+', line_clean, re.IGNORECASE):
        chapters.append((idx + 1, line_clean))
    # Look for Article headings (e.g. **Điều 1. ...**, **Điều 2. ...**, etc.)
    elif re.search(r'\*\*Điều\s+\d+', line_clean, re.IGNORECASE) or re.search(r'^Điều\s+\d+', line_clean, re.IGNORECASE):
        articles.append((idx + 1, line_clean))

print("\n--- CHAPTERS FOUND ---")
for num, chap in chapters:
    print(f"Line {num}: {chap}")
    # Print the next few non-empty lines to see titles
    next_lines = []
    i = num
    while i < len(lines) and len(next_lines) < 2:
        if lines[i].strip():
            next_lines.append(lines[i].strip())
        i += 1
    print(f"  Next lines: {next_lines}")

print("\n--- ARTICLES FOUND (First 15 and Last 5) ---")
print(f"Total articles found: {len(articles)}")
for num, art in articles[:15]:
    print(f"Line {num}: {art}")
for num, art in articles[-5:]:
    print(f"Line {num}: {art}")
